src/unstract/sdk/index.py

```diff
@@ -290,9 +293,18 @@ def index(
                 usage_kwargs=usage_kwargs,
                 process_text=process_text,
             )
+
             if not extracted_text:
                 raise IndexingError("No text available to index")
 
+            if isinstance(
+                vector_db.get_vector_db(
+                    adapter_instance_id=vector_db_instance_id, embedding_dimension=1
+                ),
+                (NoOpVectorDB),
+            ):
+                return doc_id
```

Comment:
@harini-venkataraman why do we need this check? Also why do we wait until extraction to return something we calculate before the extraction?

Reply:
@chandrasekharan-zipstack 
1. To answer to first question, right after this part of snippet, we do addition of nodes to vectorDB which has to skipped for NoOps. Hence the check. 
2. In the indexing, initialisation of embedding and its generation and x2Text NoOp is to be invoked to retian the flow as such. Hence added after the extraction.

Reply:
@harini-venkataraman why do you feel that addition of the nodes to vector DB has to be skipped for no ops?
- In my opinion we could have fractional timeouts and add a sleep for all vector DB ops. This will allow us to avoid checking for `no op` adapter type in other parts of the code

Reply:
@chandrasekharan-zipstack  The actual addition of nodes takes place in[ this part](https://github.com/Zipstack/unstract-sdk/blob/c48e21f45b92a158ab22d770a67afb5abdc0fe60/src/unstract/sdk/index.py#L310) of the code which need not be invoked for NoOps. This is because the base class invokes LLama index class method by passing the NoOp instance and this instance does not have any control.
Also, logically we dont skip for another places except for enforce types.

Reply:
@harini-venkataraman Ideally agree here that we should not need any explicit check whether it is a No-op adapter.

However, I understand that we cannot efficiently do this currently because it will need some refactoring to accommodate for the blackbox handling of LlamaIndex callback manager etc.

Hence request you to add a dev comment here surmising these points for future reference.